frontend/src/lib/user.ts

```diff
@@ -101,13 +105,56 @@ export async function register(password: string, passwordStrength: number, name:
   if (!response.ok) {
     const { errors } = await response.json() as RegisterResponseErrors;
     if (!errors) throw new Error('Missing error on non-ok response');
-    return { error: { turnstile: 'TurnstileToken' in errors, accountExists: 'Email' in errors } };
+    return { error: { turnstile: 'TurnstileToken' in errors, accountExists: 'Email' in errors, invalidInput: 'Required' in errors } };
   }
 
   const responseJson = await response.json() as JwtTokenUser;
   const userJson: LexAuthUser = jwtToUser(responseJson);
   return { user: userJson };
 }
+export function register(password: string, passwordStrength: number, name: string, email: string, locale: string, turnstileToken: string): Promise<RegisterResponse> {
+  return createUser('/api/User/registerAccount', password, passwordStrength, name, email, locale, turnstileToken);
+}
+export function acceptInvitation(password: string, passwordStrength: number, name: string, email: string, locale: string, turnstileToken: string): Promise<RegisterResponse> {
+  return createUser('/api/User/acceptInvitation', password, passwordStrength, name, email, locale, turnstileToken);
+}
+export async function createGuestUserByAdmin(password: string, passwordStrength: number, name: string, email: string, locale: string, _turnstileToken: string): Promise<RegisterResponse> {
+  const passwordHash = await hash(password);
+  const gqlInput: CreateGuestUserByAdminInput = {
+    passwordHash,
+    passwordStrength,
+    name,
+    locale,
+  };
+  if (email.includes('@')) {
+    gqlInput.email = email;
```

Comment:
Same concerns would apply to the bulk-add tool once we decided to allow email addresses: we add the account and then send a verification email. This doesn't add anything new; if we need to make a change, we'll need to change how we handle the bulk-add tool as well. I'll open a separate issue to track that, because making a change to the bulk-add tool feels like scope creep for this specific PR.

Reply:
Created https://github.com/sillsdev/languageforge-lexbox/issues/815 to track this. Leaving this conversation open for now so Kevin and/or Tim can easily respond.